content/en/blog/_posts/2020-05-14-my-exciting-journey-into-kubernetes-history.md

```diff
@@ -0,0 +1,656 @@
+[56]: https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
+
+## Labels, Labels, Labels
+
+Issues and pull requests in Kubernetes get different labels applied during its
+life-cycle. They are usually grouped via a single slash (`/`). For example, we
+have `kind/bug` and `kind/api-change` or `sig/node` and `sig/network`. An easy
+way to understand which label groups exist and how they’re distributed across
+the repository is to plot them into a bar chart:
+
+```
+> ./main analyze --labels-by-group
+```
+
+![labels-by-group-all-top-25](https://raw.githubusercontent.com/kubernetes-analysis/kubernetes-analysis/master/assets/labels-by-group-all-top-25.svg)
+
+It looks like that `sig/`, `kind/` and `area/` labels are pretty common.
+Something like `size/` can be ignored for now because these labels are
+automatically applied based on the amount of the code changes for a pull
+request. We said that we want to focus on release notes as input data, which
+means that we have to check out the distribution of the labels for the PRs. This
+means that the top 25 labels on pull requests are:
+
+```
+> ./main analyze --labels-by-name --pull-requests
+```
+
+![labels-by-name-pull-requests-top-25](https://raw.githubusercontent.com/kubernetes-analysis/kubernetes-analysis/master/assets/labels-by-name-pull-requests-top-25.svg)
+
+Again, we can ignore labels like `lgtm` (looks good to me), because every PR
+which now should get merged has to look good. Pull requests containing release
+notes automatically get the `release-note` label applied, which enables further
+filtering more easily. This does not mean that every PR containing that label
+also contains the release notes block. The label could have been applied
+manually and the parsing of the release notes block did not exist since the
+beginning of the project. This means we will probably loose a decent amount of
+input data on one hand. On the other hand we can focus on the highest possible
+data quality, because applying labels the right way needs some enhanced maturity
+of the project and its contributors.
+
+From a label group perspective I have chosen to focus on the `kind/` labels.
+Those labels are something which has to be applied manually by the author of the
+PR, they are available on a good amount of pull requests and they’re related to
+user-facing changes as well. Besides that, the `kind/` choice has to be done for
+every pull request because it is part of the PR template.
+
+Alright, how does the distribution of those labels look like when focusing only
+on pull requests which have release notes?
+
+```
+> ./main analyze --release-notes-stats
+```
+
+![release-notes-stats](https://raw.githubusercontent.com/kubernetes-analysis/kubernetes-analysis/master/assets/release-notes-stats.svg)
+
+Interestingly, we have approximately 7.000 overall pull requests containing
+release notes, but only ~5.000 have a `kind/` label applied. The distribution of
```

Comment:
nit: English locales usually use comma as the thousands separator.

Reply:
Okay, fixed that everywhere. :+1: